### PR TITLE
[Draft] Adding docs, scripts etc. for ProblemChild

### PR DESCRIPTION
**Pull request by ajosh0504**
This PR adds all the supplemental documents and scripts that will be mentioned in the ProblemChild blog. The official documentation that will go along with the release package in the detection-rules repo will also reference these docs.

### CODE REVIEW

File: Machine Learning/ProblemChild/datafeeds/datafeed-experimental-high-sum-by-host-problemchild.txt [hunk 1]
```diff
@@ -0,0 +1,54 @@
+PUT _ml/datafeeds/experimental-high-sum-by-host-problemchild
```
Comment (bfilar) [inline Machine Learning/ProblemChild/datafeeds/datafeed-experimental-high-sum-by-host-problemchild.txt:1]: Remove `PUT` statements from `datafeed` files.

change file extension from `txt` to `json`

File: Machine Learning/ProblemChild/datafeeds/datafeed-experimental-high-sum-by-parent-problemchild.txt [hunk 2]
```diff
@@ -0,0 +1,54 @@
+PUT _ml/datafeeds/experimental-high-sum-by-parent-problemchild
```
Comment (bfilar) [inline Machine Learning/ProblemChild/datafeeds/datafeed-experimental-high-sum-by-parent-problemchild.txt:1]: Remove `PUT` statements from `datafeed` files.

change file extension from `txt` to `json`

File: Machine Learning/ProblemChild/datafeeds/datafeed-experimental-high-sum-by-user-problemchild.txt [hunk 3]
```diff
@@ -0,0 +1,54 @@
+PUT _ml/datafeeds/experimental-high-sum-by-user-problemchild
```
Comment (bfilar) [inline Machine Learning/ProblemChild/datafeeds/datafeed-experimental-high-sum-by-user-problemchild.txt:1]: Remove `PUT` statements from `datafeed` files.

change file extension from `txt` to `json`

File: Machine Learning/ProblemChild/datafeeds/datafeed-experimental-rare-process--by-parent-problemchild.txt [hunk 4]
```diff
@@ -0,0 +1,54 @@
+PUT _ml/datafeeds/experimental-rare-process-by-parent-problemchild
```
Comment (bfilar) [inline Machine Learning/ProblemChild/datafeeds/datafeed-experimental-rare-process--by-parent-problemchild.txt:1]: Remove `PUT` statements from `datafeed` files.

change file extension from `txt` to `json`

File: Machine Learning/ProblemChild/datafeeds/datafeed-experimental-rare-process-by-host-problemchild.txt [hunk 5]
```diff
@@ -0,0 +1,54 @@
+PUT _ml/datafeeds/experimental-rare-process-by-host-problemchild
```
Comment (bfilar) [inline Machine Learning/ProblemChild/datafeeds/datafeed-experimental-rare-process-by-host-problemchild.txt:1]: Remove `PUT` statements from `datafeed` files.

change file extension from `txt` to `json`

File: Machine Learning/ProblemChild/datafeeds/datafeed-experimental-rare-process-by-user-problemchild.txt [hunk 6]
```diff
@@ -0,0 +1,54 @@
+PUT _ml/datafeeds/experimental-rare-process-by-user-problemchild
```
Comment (bfilar) [inline Machine Learning/ProblemChild/datafeeds/datafeed-experimental-rare-process-by-user-problemchild.txt:1]: Remove `PUT` statements from `datafeed` files.

change file extension from `txt` to `json`

File: Machine Learning/ProblemChild/job_configs/experimental-high-sum-by-host-problemchild.txt [hunk 7]
```diff
@@ -0,0 +1,25 @@
+PUT _ml/anomaly_detectors/experimental-high-sum-by-host-problemchild
```
Comment (bfilar) [inline Machine Learning/ProblemChild/job_configs/experimental-high-sum-by-host-problemchild.txt:1]: Remove `PUT` statements from `json` files.

change file extension from `txt` to `json`

File: Machine Learning/ProblemChild/job_configs/experimental-high-sum-by-parent-problemchild.txt [hunk 8]
```diff
@@ -0,0 +1,25 @@
+PUT _ml/anomaly_detectors/experimental-high-sum-by-parent-problemchild
```
Comment (bfilar) [inline Machine Learning/ProblemChild/job_configs/experimental-high-sum-by-parent-problemchild.txt:1]: Remove `PUT` statements from `json` files.

change file extension from `txt` to `json`

File: Machine Learning/ProblemChild/job_configs/experimental-high-sum-by-user-problemchild.txt [hunk 9]
```diff
@@ -0,0 +1,25 @@
+PUT _ml/anomaly_detectors/experimental-high-sum-by-user-problemchild
```
Comment (bfilar) [inline Machine Learning/ProblemChild/job_configs/experimental-high-sum-by-user-problemchild.txt:1]: Remove `PUT` statements from `json` files.

change file extension from `txt` to `json`

File: Machine Learning/ProblemChild/job_configs/experimental-rare-process-by-host-problemchild.txt [hunk 10]
```diff
@@ -0,0 +1,25 @@
+PUT _ml/anomaly_detectors/experimental-rare-process-by-host-problemchild
```
Comment (bfilar) [inline Machine Learning/ProblemChild/job_configs/experimental-rare-process-by-host-problemchild.txt:1]: Remove `PUT` statements from `json` files.

change file extension from `txt` to `json`

File: Machine Learning/ProblemChild/job_configs/experimental-rare-process-by-parent-problemchild.txt [hunk 11]
```diff
@@ -0,0 +1,25 @@
+PUT _ml/anomaly_detectors/experimental-rare-process-by-parent-problemchild
```
Comment (bfilar) [inline Machine Learning/ProblemChild/job_configs/experimental-rare-process-by-parent-problemchild.txt:1]: Remove `PUT` statements from `json` files.

change file extension from `txt` to `json`

File: Machine Learning/ProblemChild/job_configs/experimental-rare-process-by-user-problemchild.txt [hunk 12]
```diff
@@ -0,0 +1,25 @@
+PUT _ml/anomaly_detectors/experimental-rare-process-by-user-problemchild
```
Comment (bfilar) [inline Machine Learning/ProblemChild/job_configs/experimental-rare-process-by-user-problemchild.txt:1]: Remove `PUT` statements from `json` files.

change file extension from `txt` to `json`

File: Machine Learning/ProblemChild/features_endgame.txt [hunk 13]
```diff
@@ -0,0 +1,92 @@
+POST _scripts/features_endgame
+{
+  "script": {
+    "lang": "painless",
+    "source": """
+      if (ctx.containsKey('process') && ctx['process'].containsKey('name') && !ctx['process']['name'].empty){
+        ctx.feature_process_name = ctx['process']['name']
+      }
+      else{
```
Comment (bfilar) [inline Machine Learning/ProblemChild/features_endgame.txt:9]: ensure proper spacing for code brackets e.g., `else {`

**Comment by ajosh0504** [comment]
As suggested by you @bfilar , I have removed the PUTs and changed the formats of all the files to json.

**Pull request merged**